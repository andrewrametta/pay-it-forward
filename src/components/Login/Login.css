.login-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 0px 10px;
}

.demo-credentials {
  color: #5219c9;
  margin: 2.5px 0px;
}

form.login-form > div {
  margin-top: 10px;
}

@media all and (min-width: 600px) {
  div.submit-button {
    text-align: center;
  }

  button.form-btn {
    width: 30%;
  }
}
