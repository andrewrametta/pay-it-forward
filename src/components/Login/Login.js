import React, { useState, useContext } from "react";
import TokenService from "../../services/token-service";
import AuthApiService from "../../services/auth-api-service";
import AppContext from "../../AppContext";
import "./Login.css";

function Login(props) {
  const { error, setError } = useState("null");
  const context = useContext(AppContext);

  const handleLogin = (e) => {
    e.preventDefault();
    const { username, password } = e.target;
    const user = { username: username.value, password: password.value };
    setError({ error: null });
    AuthApiService.loginUser(user)
      .then((loginResponse) => {
        // store auth token in local storage
        TokenService.saveAuthToken(loginResponse.authToken);
        context.setUserType(loginResponse.authToken.type);
        props.history.push("/dashboard");
      })
      .catch((res) => {
        setError({ error: res.error });
      });
  };

  return (
    <div className="login-wrapper">
      <section className="login-section">
        <header className="login-header">
          <h1>Login</h1>
        </header>
        <article>
          <div>
            <form
              className="login form"
              aria-label="login-form"
              onSubmit={handleLogin}
            >
              <div>
                <label className="label" htmlFor="username">
                  Username
                </label>
                <input
                  type="text"
                  name="username"
                  id="username"
                  defaultValue="username"
                />
              </div>
              <div>
                <label className="label" htmlFor="password">
                  Password
                </label>
                <input
                  type="password"
                  name="password"
                  id="password"
                  defaultValue="password"
                />
              </div>
              <div className="login-button">
                <button>Login</button>
              </div>
            </form>
          </div>
        </article>
      </section>
    </div>
  );
}

export default Login;
