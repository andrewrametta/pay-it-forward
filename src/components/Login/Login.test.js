import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Login from "./Login";
// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AppContext from "../../AppContext";

configure({ adapter: new Adapter() });

describe(`Login component`, () => {
  it.only("renders the complete page", () => {
    const wrapper = shallow(
      <AppContext.Provider>
        <Login />
      </AppContext.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
