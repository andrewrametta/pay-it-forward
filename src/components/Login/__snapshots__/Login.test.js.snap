// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders the complete page 1`] = `<Login />`;
